src/bitcoin-tx.cpp

```diff
@@ -766,8 +767,6 @@ static std::string readStdin()
     if (ferror(stdin))
         throw std::runtime_error("error reading stdin");
 
-    boost::algorithm::trim_right(ret);
```

Comment:
Even though it does make sense to trim all whitespace from a hex-encoded transaction, it still also makes sense to trim newline characters here (which should be locale-dependent).

Reply:
@luke-jr I'm not sure I follow: could you explain why? Please include an example if possible where the new code would give bad results.

Reply:
Wait, how are newline characters locale dependent?